src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -66,21 +67,13 @@ export class ItemChildrenEditComponent implements OnChanges {
   private subscription?: Subscription;
   @Output() childrenChanges = new EventEmitter<PossiblyInvisibleChildData[]>();
 
-  constructor(
-    private getItemChildrenService: GetItemChildrenService,
-    private itemRouter: ItemRouter,
-  ) {}
-
-  ngOnChanges(): void {
-    this.reloadData();
-  }
-
-  reloadData(): void {
-    if (this.itemData?.currentResult) {
-      this.state = 'loading';
-      this.subscription?.unsubscribe();
-      this.subscription = this.getItemChildrenService
-        .getWithInvisibleItems(this.itemData.item.id, this.itemData.currentResult.attemptId)
+  private readonly params$ = new ReplaySubject<{ id: string, attemptId: string }>(1);
+  private readonly refresh$ = new Subject<void>();
+  readonly state$: Observable<FetchState<PossiblyInvisibleChildData[]>> = this.params$.pipe(
+    distinctUntilChanged((a, b) => a.id === b.id && a.attemptId === b.attemptId),
+    switchMap(({ id, attemptId }) =>
+      this.getItemChildrenService
+        .getWithInvisibleItems(id, attemptId)
         .pipe(
           map(children => children
```

Comment:
Same remark as upper, the `map` can be declared after `switchMap` and not inside
(Unless I'm missing something)